### ISSUE TITLE
Expand Prometheus Metrics to state device/folder names

### ISSUE THREAD

**Issue by simonmicro** [syncthing/syncthing#9519]
### Feature description

Currently, metrics like:

- syncthing_protocol_recv_bytes_total
- syncthing_model_folder_processed_bytes_total

State only the object ids as label to be scraped. I would welcome the addition of the clear-text values as labels in addition to the ids.

### Problem or use case

After scraping a Syncthing server using Prometheus, I can display the metrics with tools like Grafana. Sadly, I can only display the unresolved ids of the folders / devices inside the UI, as no further information is available.

This leads to problems that alerts cannot contain easy-to-interepret descriptions - for admins to decide if they want to start investigation. In addition, admins cannot interpret e.g. if spikes in traffic are legitimate (as the device is only visible as ID) in Grafana, which makes anomaly-investigations harder.

Finally, with growing numbers of devices and folders connected to an instance (for me `>40` folders / `>8` devices) the metrics get more and more hard to read - pie charts containing statistics about specific folders in comparison to each other are just unreadable.

Here are some samples (blue = mask for privacy):

![Screenshot from 2024-04-26 18-18-52](https://github.com/syncthing/syncthing/assets/24966116/f1faa737-9dde-42aa-b79d-5db5476493ca)
![Screenshot from 2024-04-26 18-19-03](https://github.com/syncthing/syncthing/assets/24966116/1d621cab-2287-4ac2-bb80-4e3eb250c8bc)

### Alternatives or workarounds

- Write an own exporter to scrape Syncthing-metrics and label them with the metadata available over the api.
- Relabel all ids by hand in the Prometheus config?

**Comment by SophisticaSean** [comment]
@simonmicro could you provide an example of what device metadata you'd like to see exposed? I'm thinking about giving this a shot. :)

**Comment by simonmicro** [comment]
@SophisticaSean Thank you very much about your motivation to pick this up! Mostly, I just would like to see the device's assigned name or (in case of folders) the assigned name from the web-ui. This would make identifying the metrics much easier!

But, as you ask... Maybe there is more metadata available? Could you compile a short list of ideas (be it just code references), where I could get an idea from what to suggest?

Thanks in advance!

**Comment by SophisticaSean** [comment]
> @SophisticaSean Thank you very much about your motivation to pick this up! Mostly, I just would like to see the device's assigned name or (in case of folders) the assigned name from the web-ui. This would make identifying the metrics much easier!
> 
> But, as you ask... Maybe there is more metadata available? Could you compile a short list of ideas (be it just code references), where I could get an idea from what to suggest?
> 
> Thanks in advance!

Yeah I'll take a look. My cursory glance revealed that this metric output takes the device ID's in as an argument, so we'd have to do some reworking to expose more data to the resulting prometheus metrics output.

**Comment by calmh** [comment]
I suspect there are a number of places exporting things by ID where the name is not readily available. An other option could be to expose and info metric, like a gauge with ID, name, type labels and just the value 1 for each folder & device. You can then join with that in the promql query.
